### ISSUE TITLE
mode 7 bale collector - modconflict (modhub)

### ISSUE THREAD

**Issue by rgoe** [Courseplay/courseplay#6938]
Hi there,
I have encountered a conflict of the new mode7 with a mod from the modhub - it is about the "John Deere 1110 1210 1510 G" (https://farming-simulator.com/mod.php?lang=us&country=de&mod_id=190621).
As soon as the vehicle is purchased Courseplay stops mode7 immediately after starting it:
`Exception, stopping Courseplay driver, C:/Users/UserX/Documents/My Games/FarmingSimulator2019/mods/FS19_Courseplay/vehicles.lua:1560: attempt to perform arithmetic on field 'maxTurningRadius' (a nil value)`
If the vehicle is sold again, the problem is solved and mode7 works again immediately.

Tested with
- Mods: only Courseplay, EasyDevcontrols and the above mentioned John Deere
- DLCs: all installed but none activated in the savegame
- Maps: Felsbrunn and Ravenport (and HoF 2021)
- Vehicle combination used to test: Steyr Profi CVT & Arcusin FSX6372 (both vanilla from basegame)
  (additionally also different combinations of tractors and bale collectors from the Anderson DLC with same result)
Logfile with dbg-channels 7&14: 
[log.txt](https://github.com/Courseplay/courseplay/files/6072924/log.txt)

Best Regards

**Comment by Tensuko** [comment]
Well, that is more a mod Issue then a conflict.
What you could try is to add the vehicle.xml into the config/VehicleConfigurations.xml
and add the Line, while turnRadius could also be higher, as you like.
I am not sure if this is working already, the .xml could be also in your My Games/FarmingSimulator19 folder, there is a folder named modSettings and inside there should be the same file free to edit.
I you can test this and it is working, let us know.

```
<Vehicle name="XXX.xml"
             turnRadius = "9"
    />
```

**Comment by rgoe** [comment]
Thanks for your prompt reply, unfortunately that doesn't do the trick.
Allthough I'm not a coder (at least not lua) I dug into the function where courseplay bailed out and found that the parameter (maxTurnRadius) is not verified before use (resulting in the nil value) and the parameter from VehicleConfigurations.xml not used at all. I did some changes to vehicles.lua to catch this issue. I'd try to make a pull request (my first one - so hopefully I got it right). Could you have a look into it?

**Comment by Tensuko** [comment]
That might be related to this:
#6944

**Comment by rgoe** [comment]
Solved with version [v6.03.00048](https://github.com/Courseplay/courseplay/releases/tag/6.03.00048) so I'm closing this here.
Thanks to all.
